tools/ab_test.py

```diff
@@ -109,22 +105,30 @@ def find_unit(emf: dict, metric: str):
     return metrics.get(metric, "None")
 
 
-def load_data_series(revision: str):
+def load_data_series(report_path: Path, revision: str = None, *, reemit: bool = False):
     """Loads the data series relevant for A/B-testing from test_results/test-report.json
     into a dictionary mapping each message's cloudwatch dimensions to a dictionary of
     its list-valued properties/metrics.
 
-    Also reemits all EMF logs."""
+    If `reemit` is True, it also reemits all EMF logs to a local EMF agent."""
     # Dictionary mapping EMF dimensions to A/B-testable metrics/properties
     processed_emf = {}
 
-    report = json.loads(Path("test_results/test-report.json").read_text("UTF-8"))
+    report = json.loads(report_path.read_text("UTF-8"))
     for test in report["tests"]:
         for line in test["teardown"]["stdout"].splitlines():
```

Comment:
Do we have any tests that emit metrics during "call" phase? Could we collect both phases?

Reply:
We do not currently, because all metrics are emitted by the `metrics` fixture, which only flushes during teardown.

Reply:
I've had a quick look at whether it'd be an easy change to consider both "teardown" and "call", but it seems that either the "call" key or the "stdout" subkey are not always present :(